extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -154,6 +156,14 @@ protected function checkPermissions() {
 		}
 	}
 
+	protected function checkPermissionsForJA() {
```

Comment:
In case of wikis with anonymous editing disabled wouldn't it be enough to call checkPermissions method? This code in checkPermissions ```$this->mUpload->verifyTitlePermissions( $this->mUser ) !== true``` perhaps should be already taking ```$wgDisableAnonymousEditing``` into consideration - if it's not then maybe it should be fixed.